### DISCUSSION TITLE
Releasing v10.1 on 19-Feb-2022?

### DISCUSSION THREAD

**Discussion by vhelin** [vhelin/wla-dx#477] [General]
Now that v10.1a has some important bug fixes and memory leak fixes, I'm thinking that I should release v10.1 in two days.

Any objections? :)

I'll try to write more tests for features recently added before releasing v10.1...

**Comment by Kroc** [comment]
Please can we have proper binary releases that include the docs? It's *very* hard to recommend WLA-DX to people when there's no way to just download it and get started with what's in the zip!

**Reply by vhelin** [reply]
Sure, I can add docs to the unsupported binary releases I host on my website. Perhaps I could add those binary releases to the GitHub's releases page as well...

**Comment by vhelin** [comment]
I've postponed the release of v10.1 by 6 days just because even today I've fixed some bugs in the new code... If nothing major gets reported in the next 7 days when I'll release v10.1 on next week's Satuday.

**Reply by Kroc** [reply]
Compiling with latest head on WIN10/VS2022 (using tools 141), I can't assemble anything as it errors trying to do something with the object file, existing or otherwise:
```
Pass 1...
LOAD_FILE_DATA: Could not open file "build/impatience.o".
````
I'll walk back through the git history and try to find where this started happening; 10.0 builds and runs fine

**Reply by vhelin** [reply]
Thanks! I think we need to get the tests to run on Azure Pipelines / Windows as well, not just on Linux...

**Reply by vhelin** [reply]
Just went and ran tests/z80/substitution_test under my Windows 10 and it worked without any errors... VS2017 here, "Platform Toolset" says this: "Visual Studio 2017 - Windows XP (v141_xp)"

**Reply by Kroc** [reply]
Could be a VS2022 thing; I've almost narrowed the exact commit down

**Reply by Kroc** [reply]
It's within this range... 
![image](https://user-images.githubusercontent.com/142650/153752054-c3e8150a-d0d7-4fd1-8564-5a5802e4e387.png)

**Reply by Kroc** [reply]
It's c3cfb15ce2cabc2b7223a92776f84879619cf051

**Reply by Kroc** [reply]
Something about that change causes "wla-gb.exe" to silently terminate (the object file error is because the linker can't find the expected object file)

**Reply by vhelin** [reply]
What if you download the latest exes built by Azure? https://dev.azure.com/villehelin0486/villehelin/_build/results?buildId=1089&view=artifacts&pathAsName=false&type=publishedArtifacts - Do these work?

**Reply by vhelin** [reply]
Funny coincidence: I can see this in the Azure Pipelines / Windows build log:

##[warning]A brownout will take place on February, 14 2:00 PM - 8:00 PM UTC to raise awareness of the upcoming windows-2016 environment removal. For more details, see https://github.com/actions/virtual-environments/issues/4312

That's tomorrow. I should change "vmImage: 'vs2017-win2016'" to something more recent... Let's see if we still can build Windows XP exes in Azure.

**Reply by Kroc** [reply]
Same problem -- this is running my build script from [Impatience](https://github.com/Kroc/Impatience) so it may actually be something in the code that's causing a silent error. But it's definitely that specific commit where the issue occurs

**Reply by vhelin** [reply]
Can you run the tests, or assemble some other project?

**Reply by Kroc** [reply]
Latest artefacts work with Sonic1-SMS-Z80, so it's definitely something in my Impatience code that's killing it :/

**Reply by vhelin** [reply]
Perhaps adding .PRINT "HELLO 1" etc. into Impatience's asm file(s) might show where assembling stops?

**Reply by vhelin** [reply]
I tried to build Impatience myself, but I get weird error messages

-bash: ./bin/rgbds/rgbgfx.exe: Permission denied

That's from Cygwin, but the same happens from an (Windows 10) Admin cmd...

**Reply by vhelin** [reply]
Ok, replaces .incbin's with .dsb's, now I get this

![image](https://user-images.githubusercontent.com/6417746/153760867-aaf1eb01-9605-4749-8a71-25220db2ffc2.png)

Let's see...

**Reply by Kroc** [reply]
Ah, that's it, but on Windows there's no error message, just silent termination. I can narrow down which file is causing it

**Reply by Kroc** [reply]
"cards.wla"

**Reply by Kroc** [reply]
It's this line here:
```
        ld      BC,     (SECTIONEND_cards.table - SECTIONSTART_cards.table)+1
```
Something about section labels

**Reply by vhelin** [reply]
What if you now try the latest sources, is the issue gone? I found a bug in the new code in v10.1a and fixed that...

**Reply by vhelin** [reply]
Yeah, that's (and my fix is) related to the code that checks if labels A and B in calculation A-B are in the same section...

**Reply by Kroc** [reply]
Ok, that fixed it! Great work!

**Reply by vhelin** [reply]
Nice, thank you for such a nice testcase! :)
